cloud/cluster/memory/cluster.go

```diff
@@ -0,0 +1,138 @@
+package memory
```

Comment:
so most Golang projects I've seen don't do nested interfaces this way.  
Its a bit weird since it forces us to alias packages when we import them.  Do we want to continue following this pattern?